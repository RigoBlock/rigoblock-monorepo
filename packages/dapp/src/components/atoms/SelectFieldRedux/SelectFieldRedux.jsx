import './SelectFieldRedux.scss'
import 'react-md/lib/Helpers'
import { Field } from 'redux-form'
import PropTypes from 'prop-types'
import React from 'react'
import SelectField from '../SelectField'

const SelectFieldRedux = ({ fieldName, fieldProps }) => {
  /* eslint-disable react/prop-types */
  const renderSelectField = ({ input, meta: { touched, error } }) => {
    return (
      <SelectField
        {...fieldProps}
        value={input.value}
        onChange={input.onChange}
        error={touched && !!error}
        errorText={error}
      />
    )
  }

  return <Field name={fieldName} component={renderSelectField} />
}

SelectFieldRedux.propTypes = {
  fieldName: PropTypes.string.isRequired,
  fieldProps: PropTypes.shape(SelectField.propTypes).isRequired
}

export default SelectFieldRedux
