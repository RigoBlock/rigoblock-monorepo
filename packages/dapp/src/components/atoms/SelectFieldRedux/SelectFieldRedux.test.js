import { shallow } from 'enzyme'
import SelectFieldRedux from './SelectFieldRedux.jsx'
import toJson from 'enzyme-to-json'

const itemList = ['item1', 2, { label: 'test', value: '1' }]

const props = {
  fieldName: 'testField',
  fieldProps: {
    id: 1,
    items: itemList,
    onChange: jest.fn()
  }
}

describe('SelectFieldRedux component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(SelectFieldRedux, props)))
    ).toMatchSnapshot()
  })
})
