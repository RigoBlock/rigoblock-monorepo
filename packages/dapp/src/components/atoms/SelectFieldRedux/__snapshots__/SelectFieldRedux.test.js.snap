// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldRedux component renders correctly 1`] = `
<Field
  component={[Function]}
  name="testField"
/>
`;
